instrumentation/jms/jms-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jms/v3_0/JmsSingletons.java

```diff
@@ -42,5 +47,30 @@ public static Instrumenter<MessageWithDestination, Void> consumerProcessInstrume
     return CONSUMER_PROCESS_INSTRUMENTER;
   }
 
+  public static void createReceiveSpan(
+      MessageWithDestination request, Timer timer, Throwable throwable) {
+    ContextPropagators propagators = GlobalOpenTelemetry.getPropagators();
+    boolean receiveInstrumentationEnabled =
+        ExperimentalConfig.get().messagingReceiveInstrumentationEnabled();
+    Context parentContext = Context.current();
+    if (!receiveInstrumentationEnabled) {
+      parentContext =
+          propagators
+              .getTextMapPropagator()
+              .extract(parentContext, request, MessagePropertyGetter.INSTANCE);
+    }
+
+    if (consumerReceiveInstrumenter().shouldStart(parentContext, request)) {
+      InstrumenterUtil.startAndEnd(
+          consumerReceiveInstrumenter(),
+          parentContext,
+          request,
+          null,
+          throwable,
+          timer.startTime(),
+          timer.now());
+    }
+  }
```

Comment:
This is duplicated between the 3.0 and 1.1 jms instrumentations. Can it live in common and be reused (by passing the instrumenter to a common method)?